kweeuhree/react-website
# Implement ErrorBoundary component

## Description
This pull request introduces an `ErrorBoundary` component to enhance error handling in the application. The `ErrorBoundary` component catches JavaScript errors in its child components and displays a fallback UI instead of crashing the entire application. This change aims to improve user experience by gracefully handling errors.

## Changes Made
- Created a new `ErrorBoundary.jsx` component.
- Integrated the `ErrorBoundary` into the main application layout to wrap key components.
- Implemented a fallback UI that informs users when an error occurs.

## Motivation and Context
As developers, it is crucial to handle errors gracefully in order to maintain a smooth user experience. By implementing the `ErrorBoundary`, we can ensure that users are notified of issues without causing the application to crash. This aligns with our goal of providing a robust and user-friendly interface.

## How to Test
1. Use the application and intentionally cause an error in any of the wrapped components.
2. Verify that the fallback UI is displayed instead of the broken component.
3. Ensure that the rest of the application remains functional after the error occurs.

## Related Issues
- Closes #2
